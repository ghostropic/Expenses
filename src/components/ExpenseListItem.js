import React from 'react'
import moment from 'moment'

import { StyledLink } from '../styles/style'

const toDollars = (pennies) => {
  const dollars = pennies / 100;
  return dollars.toLocaleString('en-US', { style: 'currency', currency: 'USD' });
}

const toDate = createdAt => moment(createdAt).format('MM/DD/YYYY h:mm a')

const ExpenseListItem = ({
  id,
  description,
  amount,
  note,
  createdAt
}) => (
  <div>
    <p>{toDollars(amount)}</p>
    <StyledLink to={`/edit/${id}`}>{description}</StyledLink>
    <p>{note}</p>
    <p>{toDate(createdAt)}</p>
    <hr />
  </div>
)


export default ExpenseListItem
