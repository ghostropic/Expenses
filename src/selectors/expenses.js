import moment from 'moment'

// getVisibleExpenses - passes expense and filters and determines what should be send along the expense list

export default (expenses, {text, sortBy, startDate, endDate}) => { // two arguments: expenses and destructured filters
  return expenses.filter((expense) => {
    const createdAtMoment = moment(expense.createdAt)
    const startDateMatch = startDate ? startDate.isSameOrBefore(createdAtMoment, 'day') : true
    const endDateMatch = endDate ? endDate.isSameOrAfter(createdAtMoment, 'day') : true
    // const endDateMatch = typeof endDate !== 'number' || expense.createdAt <= endDates
    const textMatch = text === '' || expense.description.toLowerCase().includes(text.toLowerCase())
    return startDateMatch && endDateMatch && textMatch
  }).sort((a,b) => {
    if (sortBy === 'date') {
      return a.createdAt < b.createdAt ? 1 : -1
    } else if (sortBy === 'amount') {
      return a.amount < b.amount ? 1 : -1
    }
  })
  // return [expenses, startDate]
}