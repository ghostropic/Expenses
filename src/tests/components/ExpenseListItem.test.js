import React from 'react'
import { shallow } from 'enzyme'
import moment from 'moment'

import ExpenseListItem from '../../components/ExpenseListItem'
import expenses from '../fixtures/expenses';

test('should render list of expenses', () => {
  // const date = moment(wtf[0].createdAt).format('MM/DD/YYYY h:mm a')

  const wrapper = shallow(<ExpenseListItem {...expenses[0]} />)

  expect(wrapper).toMatchSnapshot()
})
 