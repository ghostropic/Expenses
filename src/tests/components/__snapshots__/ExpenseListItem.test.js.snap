// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render list of expenses 1`] = `
<div>
  <p>
    $1.95
  </p>
  <style__StyledLink
    to="/edit/1"
  >
    gum
  </style__StyledLink>
  <p>
    one pack of bubble gum
  </p>
  <p>
    12/31/1969 7:00 pm
  </p>
  <hr />
</div>
`;
